api/agent.go

```diff
@@ -395,15 +396,30 @@ func NewControllerWrapper(config *loadtest.Config, controllerConfig interface{},
 			username = creds[id].username
 			email = creds[id].email
 			password = creds[id].password
+
+			// Check if the user has a custom authentication type. Custom authentication types are
+			// specified by prepending the username with the authentication type followed by a colon.
+			// Example: "openid:user1@test.mattermost.com user1password"
+			if usernameParts := strings.Split(username, ":"); len(usernameParts) > 1 {
+				authenticationType = usernameParts[0]
+				username = usernameParts[1]
+
+				// Fix the email as well
+				if emailParts := strings.Split(email, ":"); len(emailParts) > 1 {
+					email = emailParts[1]
+				}
+			}
```

Comment:
I think this could become a util function.

Reply:
I'm going to make a note of this but refrain from modifying too much logic until this is merged now that we have this working. I can address the rest of the comments in individual PRs later.